### PR TITLE
core-image-pelux: add git to dev images

### PR DESCRIPTION
**Pull request by sashko**
Signed-off-by: Oleksandr Kravchuk <oleksandr.kravchuk@pelagicore.com>

### CODE REVIEW

**Comment by rpannek** [comment]
This question should have been answered in the commit message, because people in the future will have it too.

**Comment by thenor** [comment]
The biggest reason for my skepticism is that we have been discussing that we want to make the images more lightweight and find what is included but not needed. If this is in fact needed for a specific use case, then that's one thing, but if this is just "would be nice to have" then I don't see any urgency and it can wait until a use case presents itself or at least until after the analysis of the image (an removal of unnecessary includes) is done.

**Comment by rpannek** [comment]
Wait, so there was no discussion about it and it wasn't even attempted to be explained in the commit message? How in the world are people supposed to review something like that??

**Comment by JEderonn** [comment]
As a developer, don't you develop on your own machine, using the SDK to cross-compile, and then deploy the artifacts to the target and test. (Or bitbake the complete image and deploy)

Has the team defined what a dev-image mean? I would assume that it has the same content as the non-dev image but with libraries with debug symbols, gdb, valgrind and ssh deployed to target. 

In the handbook it' stated: "Both PELUX images are available as -dev variants, which include some extra development and debugging tools. For some platforms, those images are also available as -update to generate update artifacts for the SOTA System."
Not so much, so perhaps there is a need to add more info about what a dev image is?

To summarize, I think there is a need to discuss this in the team before this pull request can be merged.

**Comment by thenor** [comment]
> As a developer, don't you develop on your own machine, using the SDK to cross-compile, and then deploy the artifacts to the target and test. (Or bitbake the complete image and deploy)
> 
I personally don't see the need to develop on the Pelux image. I would develop on a separate machine and then recompile. I guess if you just want a specific package added on your Pelux image you could use scp.

> Has the team defined what a dev-image mean? I would assume that it has the same content as the non-dev image but with libraries with debug symbols, gdb, valgrind and ssh deployed to target.
> 
> In the handbook it' stated: "Both PELUX images are available as -dev variants, which include some extra development and debugging tools. For some platforms, those images are also available as -update to generate update artifacts for the SOTA System."
> Not so much, so perhaps there is a need to add more info about what a dev image is?
> 
> To summarize, I think there is a need to discuss this in the team before this pull request can be merged.
I agree that this most likely needs to be defined so that the entire team is aware of the purpose and use cases of the different images.
